ddtrace/debugging/README.md

```diff
@@ -0,0 +1,184 @@
+# Datadog Debugging
+
+This area contains the implementation of the following products:
+
+- Dynamic Instrumentation
+- Exception Replay
+- Code Origin for Spans
+
+
+## General Architectural Overview
+
+Many of the products listed above need to generate signals, most often in the
+form of snapshots. These are special log messages that contain a log messages,
+and potentially extra debugging information, such as the value of local
+variables, arguments, return value, exceptions, stack traces, etc.
+
+The unit responsible for uploading the generated snapshots is the
+`LogsIntakeUploaderV1`. This is a periodic worker thread that monitors a queue
+of encoded snapshots that are ready to be uploaded to the logs intake via the
+Datadog agent. Products needing to upload snapshots must register with the
+`LogsIntakeUploaderV1`. This will ensure that the uploader is started/stopped as
+required.
+
+```plantuml
```

Comment:
Is PlantUML the default for this repository? If you want to have these natively render in Github markdown, you could use the [mermaid diagram format](https://github.blog/developer-skills/github/include-diagrams-markdown-files-mermaid/), for example:

```mermaid
flowchart LR
    subgraph LogsIntakeUploaderV1
        r("register (portin)")
        e("enqueue (portin)")
        o("HTTP (portout)")
    end

    P("Product") --> r
    P -.-> e
    o --> A("Datadog Agent")
```

Reply:
> Is PlantUML the default for this repository?

I think this would be the first example of in-repo architecture docs, so probably whatever we go for here will become the standard 🙂 

> If you want to have these natively render in Github markdown, you could use the [mermaid diagram format](https://github.blog/developer-skills/github/include-diagrams-markdown-files-mermaid/)

I found out that GH has baked-in support for Mermaid after I created the diagrams 😭 . I'll try to see if I can migrate them over.